homeassistant/components/nightscout/sensor.py

```diff
@@ -28,10 +45,37 @@ async def async_setup_entry(
     entry: ConfigEntry,
     async_add_entities: AddEntitiesCallback,
 ) -> None:
-    """Set up the Glucose Sensor."""
+    """Set up entities."""
     api = hass.data[DOMAIN][entry.entry_id]
+
+    # Glucose sensor
     async_add_entities([NightscoutSensor(api, "Blood Sugar", entry.unique_id)], True)
 
+    # Uploader batteries
+    async def async_update_batteries():
+        """Fetch the latest data from Nightscout REST API and update the state of devices batteries."""
+        try:
+            return await api.get_latest_devices_status()
+        except OSError as error:
+            _LOGGER.error(
+                "Error fetching battery devices status. Failed with %s", error
+            )
+            raise UpdateFailed(f"Error communicating with API: {error}") from error
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name="battery_sensor",
+        update_method=async_update_batteries,
+        update_interval=timedelta(minutes=1),
+    )
+
+    await coordinator.async_config_entry_first_refresh()
```

Comment:
We can only use this method during config entry setup in the init module, not here in the platform. Use this instead or more the coordinator to the init module.

https://github.com/home-assistant/core/blob/9971831b51f96ad3f1fc5f0307082c39ec5fb2dd/homeassistant/helpers/update_coordinator.py#L163